lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -341,31 +341,80 @@ private VerifiableCredentialBundle getVerifiableCredentials(
+            VerifiableCredentialBundle vcBundle = null;
             if (isNullOrEmpty(evcsIdentityVcs)) {
                 evcsIdentityVcs = evcsVcs.get(CURRENT);
                 isPendingEvcs = false;
             }
-
             if (!isNullOrEmpty(evcsIdentityVcs)) {
-                return new VerifiableCredentialBundle(
-                        configService.enabled(EVCS_READ_ENABLED) ? evcsIdentityVcs : tacticalVcs,
-                        true,
-                        isPendingEvcs);
+                vcBundle =
+                        new VerifiableCredentialBundle(
+                                configService.enabled(EVCS_READ_ENABLED)
+                                        ? evcsIdentityVcs
+                                        : tacticalVcs,
+                                true,
+                                isPendingEvcs);
+
+                hasPartiallyMigratedVcs =
+                        hasPartiallyMigratedVcs(
+                                tacticalVcs,
+                                evcsIdentityVcs,
+                                isPendingEvcs,
+                                vcBundle.isF2fIdentity());
+            }
+            logIdentityMismatches(tacticalVcs, evcsVcs, hasPartiallyMigratedVcs);
+            // only use these evcs vcs if they exist and have been fully migrated
+            if (vcBundle != null && !hasPartiallyMigratedVcs) {
+                return vcBundle;
             }
         }
         return new VerifiableCredentialBundle(tacticalVcs, false, false);
     }
 
+    private boolean hasPartiallyMigratedVcs(
+            List<VerifiableCredential> tacticalVcs,
+            List<VerifiableCredential> evcsVcs,
+            boolean isPending,
+            boolean isF2f) {
+
+        if (!isPending) {
+            return false;
+        }
+        // EVCS contains only a pending F2F VC
+        if (isF2f && evcsVcs.size() == 1) {
+            return true;
+        }
+        // Tactical contains the same as pending EVCS, with one extra F2F VC
+        var extraF2fVcs =
+                tacticalVcs.stream()
+                        .filter(
+                                credential ->
+                                        F2F.getId().equals(credential.getCriId())
+                                                && evcsVcs.stream()
+                                                        .noneMatch(
+                                                                evcsVC ->
+                                                                        evcsVC.getVcString()
+                                                                                .equals(
+                                                                                        credential
+                                                                                                .getVcString())))
+                        .toList();
+
+        return extraF2fVcs.size() == 1;
```

Comment:
Do you also need to check `tacticalVcs.size() == evcsVcs.size() + 1`, in case they differ in some non-F2F VCs?

Reply:
Yes can do - wasn't sure if that was likely to happen